src/core/mempool.rs

```diff
@@ -1129,19 +1300,30 @@ impl MemPoolDB {
         Ok(updated)
     }
 
-    ///
     /// Iterate over candidates in the mempool
-    ///  `todo` will be called once for each transaction whose origin nonce is equal
-    ///  to the origin account's nonce. At most one transaction per origin will be
-    ///  considered by this method, and transactions will be considered in
-    ///  highest-fee-first order.  This method is interruptable -- in the `settings` struct, the
-    ///  caller may choose how long to spend iterating before this method stops.
+    /// `todo` will be called once for each transaction that is a valid
+    /// candidate for inclusion in the next block, meaning its origin and
+    /// sponsor nonces are equal to the nonces of the corresponding accounts.
+    /// Best effort will be made to process the transactions in fee-rate order
```

Comment:
i think it is guaranteed to work before the cache is full right?

like "Best effort will be made to process the transactions in fee-rate order. That is, transactions will be processed in fee-rate order until the candidate cache is full, ..."